File: docs-examples/example-java/src/test/java/io/micronaut/mqtt/docs/custom/type/ProductClient.java
package io.micronaut.mqtt.docs.custom.type;

// tag::imports[]
import io.micronaut.mqtt.annotation.Topic;
import io.micronaut.mqtt.v5.annotation.MqttPublisher;
import io.micronaut.context.annotation.Requires;
// end::imports[]

@Requires(property = "spec.name", value = "ProductInfoSpec")
// tag::clazz[]
@MqttPublisher
public interface ProductClient {

    @Topic("product")
    void send(byte[] data, ProductInfo productInfo);
}
// end::clazz[]

File: docs-examples/example-kotlin/src/test/kotlin/io/micronaut/mqtt/docs/custom/type/ProductClient.kt
package io.micronaut.mqtt.docs.custom.type

// tag::imports[]
import io.micronaut.mqtt.v5.annotation.MqttPublisher
import io.micronaut.mqtt.annotation.Topic
// end::imports[]

import io.micronaut.context.annotation.Requires

@Requires(property = "spec.name", value = "ProductInfoSpec")
// tag::clazz[]
@MqttPublisher
interface ProductClient {

    @Topic("product")
    fun send(productInfo: ProductInfo)

}
// end::clazz[]
